src/services/loop/loop-rest-client.ts

```diff
@@ -0,0 +1,125 @@
+// https://lightning.engineering/loopapi/#loop-rest-api-reference
+import fs from "fs"
+import https from "https"
+
+import axios, { AxiosResponse } from "axios"
+
+import { getLoopConfig } from "@config"
+
+// @todo - get from yaml or .envrc
+const loopMacaroon = convertMacaroonToHexString(
+  fs.readFileSync("./dev/lnd/loop.macaroon"),
+)
+const cert = fs.readFileSync("./dev/lnd/tls.cert")
+const key = fs.readFileSync("./dev/lnd/tls.key")
+
+export const loopRestClient = {
```

Comment:
Please implement an interface, you can use as example cache service interface (it uses local and redis with the same interface). I think in this specific case probably will include other services in the future.

Reply:
indeed, [bolt](https://boltz.exchange/) could be added as well in a future PR

Reply:
yeah i need to look at how "Ride the lightning" does this because they use both Loop and Boltz for swaps https://github.com/Ride-The-Lightning/RTL/tree/44412d357e205bbe79d14d394889851392d140c1/src/app/shared/components/services

Reply:
Loop and Boltz have slightly different api's and vocabulary but I can come up with a common interface:

Loop Service (RTL)
https://github.com/Ride-The-Lightning/RTL/blob/44412d357e205bbe79d14d394889851392d140c1/src/app/shared/services/loop.service.ts

Boltz Service (RTL)
https://github.com/Ride-The-Lightning/RTL/blob/44412d357e205bbe79d14d394889851392d140c1/src/app/shared/services/boltz.service.ts